### DISCUSSION TITLE
support for `packageManager` property in `package.json` for javascript projects

### DISCUSSION THREAD

**Discussion by travi** [renovatebot/renovate#11800] [Suggest an Idea]
the [release notes of v16.9.0 of node.js](https://github.com/nodejs/node/releases/tag/v16.9.0) highlight Corepack for helping projects specify and simplify use of the preferred package manager for the project. 

this is driven by a [`packageManager` property in the `package.json`](https://nodejs.org/dist/latest-v16.x/docs/api/packages.html#packages_packagemanager). the property enables specifying both the name of the package manager and the version. i see this as a potentially very helpful way to encourage a team to use the same version across the team, but i am wondering if that value growing stale over time could have negative impacts. 

have you considered management of this version number through renovate so that it could be updated as new versions of the [supported package managers](https://nodejs.org/dist/latest-v16.x/docs/api/corepack.html#corepack_supported_package_managers) become available? (npm is not yet supported by corepack, but is a valid value for the `packageManager` property. i would find it valuable to apply similar updates when npm is specified, in hopes that it will be added to the corepack supported managers list at some point.)

**Comment by rarkins** [comment]
Yes, I think that's a good idea and should be able to be added to the `npm` package manager in Renovate pretty easily. Alternatively it could be split into a new manager if anyone feels strongly

**Comment by HonkingGoose** [comment]
I've made a new issue to track this feature request, see:

- #11808
